libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1020,18 +1020,28 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 
 		ArrayType const* arrayType = TypeProvider::bytesMemory();
 
-		auto array = convert(*arguments[0], *arrayType);
+		if (auto const* stringLiteral = dynamic_cast<StringLiteralType const*>(arguments.front()->annotation().type))
+		{
+			// Optimization: Compute keccak256 on string literals at compile-time.
```

Comment:
Nicer would of course be if we had yul memory objects, would have the optimizer realize that the entire memory object has known contents and then have the optimizer first replace ``keccak256`` with its evaluation and then remove the memory object as unused :-)... but well...

Reply:
Yes, that would be very nice indeed! :)